python/src/iceberg/types.py

```diff
@@ -208,6 +208,10 @@ def __init__(self, *fields: NestedField, **kwargs):
             fields = kwargs["fields"]
         object.__setattr__(self, "fields", fields)
 
+    # This __post_init__ call is needed as IcebergType is not a dataclass
+    def __post_init__(self):
+        super().__init__()
```

Comment:
What happens if we call this in the init method right above? Does that interfere with setting this to a frozen dataclass?

Reply:
Yes, I believe that interferes with setting this to a frozen dataclass.

See this SO (the answer at the bottom): https://stackoverflow.com/questions/66827839/super-classs-init-is-not-called-using-dataclass

As well as this example: https://github.com/ericvsmith/dataclasses/issues/142

Reply:
To be honest, I'm not actually sure. I _think_ we might be able to safely just ignore this, but this was the suggested route so I went with it.

Reply:
For now, let's just add a single ignore statement and then come back to this so that CI is passing in the master branch.

Reply:
Just in case it ever does come up, `IcebergType ` can be protected from having frozen-dataclass-children by using `object.__setattr__` for assignments instead of `=`, e.g. instead of `self.x = x` do `object.__setattr__(self, 'x', x)`.